src/rosdiscover/config.py

```diff
@@ -2,14 +2,68 @@
 __all__ = ('Config',)
 
 from types import MappingProxyType
-from typing import Any, Mapping, Sequence
+import typing as t
 
 from loguru import logger
 import attr
 import roswire
 import yaml
 
 
+@attr.s(frozen=True, slots=True, auto_attribs=True)
+class NodeSourceInfo:
+    """
+    This is a class to represent nodes and their sources, as a bootstrap for recovering node information
+    from static analysis.
+
+    Note: This is an interim class, and will be replaced by CMakelists.txt analysis in the fullness of time.
+
+    Attributes
+    ----------
+    package_name: str
+        The name of the package where the node is from
+    node_name: str
+        The name of the node provided by the package
+    sources: Sequence[str]
+        The list of sources for building the node
+    """
+    package_name: str
+    node_name: str
+    sources: t.Sequence[str]
```

Comment:
We'll also want to specify whether the node is a Python or C++ node (e.g., via an enum).

Reply:
Let's defer this until we have the stuff from ros-anatomy, and do proper CMakelists.txt parsing. For now, because we're only integrating the C++ checking, let's just assume everything is C++